bpf/lbmap/lbmap.go

```diff
@@ -98,7 +125,12 @@ func UpdateService(key ServiceKey, value ServiceValue) error {
 }
 
 func DeleteService(key ServiceKey) error {
-	return key.Map().Delete(key.Convert())
+	err := key.Map().Delete(key.Convert())
+	if err == nil {
+		return LookupAndDeleteServiceWeights(key)
```

Comment:
Does it make sense to attempt deletion in the RR map as well even if the deletion in the main table failed? We should leave behind as little inconsistency as possible when a delete operation fails.